buildSrc/subprojects/parser/src/main/kotlin/org/apache/calcite/buildtools/parser/DialectGenerateTask.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.buildtools.parser
+
+import java.io.File
+import java.util.LinkedList
+import java.util.Queue
+import javax.inject.Inject
+import kotlin.collections.HashMap
+import kotlin.collections.MutableMap
+import kotlin.text.Regex
+import kotlin.text.StringBuilder
+import org.gradle.api.DefaultTask
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.TaskAction
+
+open class DialectGenerateTask @Inject constructor(
+    objectFactory: ObjectFactory
+) : DefaultTask() {
+
+    @InputDirectory
+    val dialectDirectory = objectFactory.directoryProperty()
```

Comment:
Would this be a constant? If so, is it reasonable to add the "const" in declaration?

Reply:
I'm not sure if kotlin has consts. The val keyword is pretty much const though as once it is assigned it can't be changed.

Reply:
I think it is still up to you if you want to make this a constant. It depends on if you would want future contributor to think twice if they want to change the value of dialectDirectory in the middle of the class.

Reply:
It shouldn't be possible to change it though since it is a val and not a var.